build/gen_stub.php

```diff
@@ -968,13 +1192,153 @@ private function appendMethodSynopsisTypeToElement(DOMDocument $doc, DOMElement
 class ClassInfo {
     /** @var Name */
     public $name;
+    /** @var int */
+    public $flags;
+    /** @var string */
+    public $type;
+    /** @var string|null */
+    public $alias;
+    /** @var bool */
+    public $isDeprecated;
+    /** @var bool */
+    public $isStrictProperties;
+    /** @var Name[] */
+    public $extends;
+    /** @var Name[] */
+    public $implements;
+    /** @var PropertyInfo[] */
+    public $propertyInfos;
     /** @var FuncInfo[] */
     public $funcInfos;
 
-    public function __construct(Name $name, array $funcInfos) {
+    /**
+     * @param Name[] $extends
+     * @param Name[] $implements
+     * @param PropertyInfo[] $propertyInfos
+     * @param FuncInfo[] $funcInfos
+     */
+    public function __construct(
+        Name $name,
+        int $flags,
+        string $type,
+        ?string $alias,
+        bool $isDeprecated,
+        bool $isStrictProperties,
+        array $extends,
+        array $implements,
+        array $propertyInfos,
+        array $funcInfos
+    ) {
         $this->name = $name;
+        $this->flags = $flags;
+        $this->type = $type;
+        $this->alias = $alias;
+        $this->isDeprecated = $isDeprecated;
+        $this->isStrictProperties = $isStrictProperties;
+        $this->extends = $extends;
+        $this->implements = $implements;
+        $this->propertyInfos = $propertyInfos;
         $this->funcInfos = $funcInfos;
     }
+
+    public function getRegistration(): string
+    {
+        $params = [];
+        foreach ($this->extends as $extends) {
+            $params[] = "zend_class_entry *class_entry_" . implode("_", $extends->parts);
+        }
+        foreach ($this->implements as $implements) {
+            $params[] = "zend_class_entry *class_entry_" . implode("_", $implements->parts);
+        }
+        foreach ($this->propertyInfos as $property) {
+            $type = $property->type;
+            if ($type === null) {
+                continue;
+            }
+
+            $arginfoType = $type->toArginfoType();
+            if (count($arginfoType->classTypes) == 1) {
+                $params[] = "zend_class_entry *class_entry_" . str_replace("\\", "_", $arginfoType->classTypes[0]->name);
+            }
+        }
+        $params = array_unique($params);
+
+        $escapedName = implode("_", $this->name->parts);
+
+        $code = "zend_class_entry *register_class_$escapedName(" . implode(", ", $params) . ")\n";
+
+        $code .= "{\n";
+        $code .= "\tzend_class_entry ce, *class_entry;\n\n";
+        if (count($this->name->parts) > 1) {
+            $className = array_pop($this->name->parts);
```

Comment:
This modified $this->name in-place, which is a bit ugly. I'd call `$this->name->getLast()` here and `$this->name->slice(0, -1)` to get the namespace.

Reply:
Uh, definitely ugly. :) I forgot about that the parameter is by-ref. I like your suggestion!